tests/unit/iontests.js

```diff
@@ -273,30 +273,46 @@ define([
             return fs.readFileSync(path, options);
         }
 
-        function roundTripEventStreams(reader) {
+        function roundTripEventStreams(input) {
             let streams = [];
-            streams.push(new es.IonEventStream(reader));
+            streams.push(new es.IonEventStream(new ion.makeReader(input)));
+            let textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(input));
+            textWriter.close();
+            let text = textWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(text)));
+            let binaryWriter = new ion.makeBinaryWriter();
+            binaryWriter.writeValues(new ion.makeReader(input));
+            binaryWriter.close();
+            let binary = binaryWriter.getBytes();
+            streams.push(new es.IonEventStream(new ion.makeReader(binary)));
+
+            //second pass
+            textWriter = new ion.makeTextWriter();
+            textWriter.writeValues(new ion.makeReader(text));
+            textWriter.close();
+            streams.push(new es.IonEventStream(new ion.makeReader(textWriter.getBytes())));
```

Comment:
I can't help but notice that there sure is a lot of copy-paste, I would hoist the logic into a generic helper.

Perhaps something like:

```javascript
function makeStream(src, makeWriter) {
  const writer = makeWriter;
  writer.writeValues(new ion.makeReader(src));
  writer.close();
  const serialized = writer.getBytes();
  return new es.IonEventStream(new ion.makeReader(serialized));
}
```

Then the call sites would be:

```javascript
// second pass over input as serialized as text...
streams.push(makeStream(text, ion.makeTextWriter));
```

I think that would make this wall of code way easier to read.

Reply:
done